### DISCUSSION TITLE
If the IsNotEmpty verification fails, would like to skip the other verifications.

### DISCUSSION THREAD

**Discussion by onozaty** [typestack/class-validator#2419] [Questions & Answers] [answered]
```ts
import { IsEmail, IsNotEmpty } from 'class-validator';

export class CreateUserDto {
  @IsNotEmpty()
  @IsEmail()
  email: string;
}
```

If an empty string is specified for email, both `@IsNotEmpty` and `@IsEmail` validation will fail.
Two messages are generated.

* email must be an email
* email should not be empty

I want to skip `IsEmail` validation in these cases, since it is sufficient to just tell the user that it is empty.

I know that this can be achieved by creating a Custom validation decorator that performs the same checks as `IsEmail` only if it is non-empty.

If there is another, simpler solution, I would like to know.

Thanks.

**Comment by sunnypatell** [comment, marked as answer]
@onozaty use the `stopAtFirstError` option when calling `validate()`. this stops validation on each property after the first constraint fails, so if `@IsNotEmpty` already caught the empty string, `@IsEmail` won't run:

```ts
const errors = await validate(dto, { stopAtFirstError: true });
```

in NestJS, pass it to `ValidationPipe`:

```ts
app.useGlobalPipes(new ValidationPipe({ stopAtFirstError: true }));
```

this is a built-in option in class-validator v0.14+. make sure your decorators are ordered the way you want (first decorator = first check).

ref: [class-validator validate options](https://github.com/typestack/class-validator#passing-options)

**Reply by onozaty** [reply]
Thank you very much.
